File: examples/array_iteration.js
const array = ['a', 'b', 'c']
const formatArray = values => `[${values.join(' ')}]`

array.forEach((value, i) => {
  console.log(i, value)
})

const mapped = array.map(value => {
  return value.toUpperCase()
})

console.log(formatArray(mapped))

const filtered = array.filter((value, i) => {
  return i % 2 == 0
})

console.log(formatArray(filtered))

const reduced = array.reduce((acc, value, i) => {
  if (i % 2 == 0) {
    acc.push(value.toUpperCase())
  }

  return acc
}, [])

console.log(formatArray(reduced))

File: examples/array_iteration.rb
array = %w[a b c]
array.each_with_index { |value, index| puts "#{index} #{value}" }

mapped = array.map(&:upcase)
puts "[#{mapped.join(' ')}]"

filtered = array.each_with_index.select { |_value, index| index.even? }.map(&:first)
puts "[#{filtered.join(' ')}]"

reduced = array.each_with_index.each_with_object([]) do |(value, index), acc|
  acc << value.upcase if index.even?
end
puts "[#{reduced.join(' ')}]"
